### PR TITLE
Add new M2 rule

### LINKED ISSUE #47: Remove the `Waiting for author` label automatically when the author or another contributor answers on the issue
### Is your feature request related to a problem?

While treating the `NMI` issues, it's so annoying to open all issues to check if the author or another contributor has answered the issue

### Describe the solution you'd like

The issue-bot should remove the `Waiting for author` label ( **only on issues**) each time the author or another contributor has responded to the issue.

1. After treating an issue, we assign the `NMI` and `Waiting for author` labels.
2. If the author or another contributor has responded to the issue, **the issue-bot should remove the `Waiting for author` label**.
3. Then, we just need to filter: `is:open is:issue label:NMI -label:"Waiting for author" ` to treat/answer on these issues.

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | This PR adds a new scenario: when an issue receives a new comment, whoever wrote it, the label `waiting for authors` is removed (if it exists)
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #47
| How to test?      | See issue #47
| Possible impacts? | --

### CODE REVIEW

File: src/ruleFinder/IssueCommentsRuleFinder.js [hunk 1]
```diff
@@ -52,14 +52,19 @@ module.exports = class IssueCommentsRuleFinder {
...
 
     if (!Utils.contextHasAction(context, 'created')) {
-      return [];
+      return rules;
     }
 
     const {issue} = context.payload;
     const repositoryConfig = this.configProvider.getRepositoryConfigFromIssue(this.config, issue);
+
+    if (Utils.issueHasLabel(issue, repositoryConfig.labels.waitingAuthor.name)) {
```
Comment (NeOMakinG) [inline src/ruleFinder/IssueCommentsRuleFinder.js:61]: Mmmh, shouldn't we verify if the comment has been added from the author?

Comment (matthieu-rolland) [reply src/ruleFinder/IssueCommentsRuleFinder.js:61]: we want to remove the label whoever wrote the comment (could be the author, anyone), at least that's what was asked, sometimes someone else gives an information that will unlock the PR

that way the issue will appear in the QA's filter: `is:open is:issue label:NMI -label:"Waiting for author"`

Comment (NeOMakinG) [reply src/ruleFinder/IssueCommentsRuleFinder.js:61]: So, what if I add the label before commenting? :trollface:

Comment (matthieu-rolland) [reply src/ruleFinder/IssueCommentsRuleFinder.js:61]: I checked with them, we should also check that the NMI label is there too

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**